### DISCUSSION TITLE
Using Tauri as a desktop environment

### DISCUSSION THREAD

**Discussion by ghost** [tauri-apps/tauri#5408] [Ideas]
A hypothetical thought for the moment but much like how KDE uses Qt, Tauri, or components from it could be used in a simillar manner, the question would be how hard would it be to do so? Now I'm not suggesting Tauri itself changes to allow for that to happen, but how it would be possible generally.  

I'm thinking about a rust built environment that uses HTML, CSS and JS for the UI for both the OS itself as well as the applications used to allow users to interact with their system (file manager, task manager, control panels) etc, but rust for everything underneath, all the way down to webkit2gtk3 and in a way become a 3rd option for developers to choose from out of the existing QT and Gnome based environments. I'm also aware of Pop OS!'s work on Cosmic as well which is an exciting project.

**Comment by Apogeum12** [comment]
Interesing idea, I thinked about some simmilar to this. The GTK Apps could run in web browser as html with broadway
More information: https://docs.gtk.org/gtk4/broadway.html <--- This could help port some app before native support in this idea.
btw. I recomend tried the Nobara project based on Fedora. A lot better then PopOs or Ubuntu.
